app/models/user.rb

```diff
@@ -8,16 +8,17 @@ class User < ApplicationRecord
   before_create :upcase_name
   validates :birthday, presence: true
   validate :birthday_within_last_100_years
-  
+  validates_confirmation_of :password_digest
+
   private
-    
-    def upcase_name
-      self.name = name.upcase
-    end
-    
-    def birthday_within_last_100_years
-      return unless birthday.present? && birthday < Settings["HUNDRED_YEARS"].years.ago.to_date
-      
-      errors.add(:birthday, :birthday_error)
-    end
+
+  def upcase_name
+    self.name = name.upcase
+  end
+
+  def birthday_within_last_100_years
+    return birthday if birthday > Settings["HUNDRED_YEARS"].years.ago.to_date
```

Comment:
Line này sữa thành logic khác, mô tả 1 chút vì sao sửa

Reply:
Cái này chị Vân bảo return some_thing if condition true á

Reply:
```suggestion
    return if birthday && birthday > Settings["HUNDRED_YEARS"].years.ago.to_date
```
Hàm này mục đích là để check validate
Nếu thỏa điều kiện, thì return luôn, hợp lệ
Nếu không thì thực hiện logic lỗi, add errors
Chỗ này thì nên dùng if thay vì unless để dễ đọc code nha